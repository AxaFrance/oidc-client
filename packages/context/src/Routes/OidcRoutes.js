import React from 'react';
import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import AuthenticationRoutes from './AuthenticationRoutes';

const propTypes = {
  notAuthenticated: PropTypes.node,
  notAuthorized: PropTypes.node,
  children: PropTypes.node,
};

const defaultProps = {
  notAuthenticated: null,
  notAuthorized: null,
  children: null,
};

const OidcRoutes = ({ notAuthenticated, notAuthorized, children }) => (
  <Switch>
    <Route
      path="/authentication"
      component={AuthenticationRoutes(notAuthenticated, notAuthorized)}
    />
    <Route render={() => children} />
  </Switch>
);

OidcRoutes.propTypes = propTypes;
OidcRoutes.defaultProps = defaultProps;

export default OidcRoutes;
