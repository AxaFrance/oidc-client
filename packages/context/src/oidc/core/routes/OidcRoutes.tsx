import React, { ComponentType, FC, PropsWithChildren, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { getPath } from './route-utils';
import CallbackComponent from '../default-component/Callback.component';
import SilentCallbackComponent from "../default-component/SilentCallback.component";
import ServiceWorkerInstall from "../default-component/ServiceWorkerInstall.component";

const propTypes = {
  callbackComponent: PropTypes.elementType,
  redirect_uri: PropTypes.string.isRequired,
  children: PropTypes.node,
};

const defaultProps: Partial<OidcRoutesProps> = {

};

type OidcRoutesProps = {
  callbackSuccessComponent?: ComponentType;
  callbackErrorComponent?: ComponentType;
  authenticatingComponent?: ComponentType;
  configurationName:string;
  redirect_uri: string;
  silent_redirect_uri?: string;
};

const OidcRoutes: FC<PropsWithChildren<OidcRoutesProps>> = ({
  callbackErrorComponent,
  callbackSuccessComponent,
                                                              authenticatingComponent,  
                                                              redirect_uri,
                                                              silent_redirect_uri,
  children, configurationName
}) => {
  // This exist because in next.js window outside useEffect is null
  const pathname = window ? window.location.pathname : '';
  
  const [path, setPath] = useState(pathname);
  
  useEffect(() => {
    const setNewPath = () => setPath(window.location.pathname);
    setNewPath();
    window.addEventListener('popstate', setNewPath, false);
    return () => window.removeEventListener('popstate', setNewPath, false);
  });
  
  const callbackPath = getPath(redirect_uri);

  if(silent_redirect_uri){
    if(path === getPath(silent_redirect_uri)){
      return <SilentCallbackComponent configurationName={configurationName} />
    }
  }

  switch (path) {
    case callbackPath:
      return <CallbackComponent callBackError={callbackErrorComponent} callBackSuccess={callbackSuccessComponent} configurationName={configurationName} />;
    case callbackPath +"/service-worker-install" :
      return <ServiceWorkerInstall callBackError={callbackErrorComponent} authenticating={authenticatingComponent} configurationName={configurationName} />;  
    default:
      return <>{children}</>;
  }
};

// @ts-ignore
OidcRoutes.propTypes = propTypes;
OidcRoutes.defaultProps = defaultProps;

export default React.memo(OidcRoutes);
