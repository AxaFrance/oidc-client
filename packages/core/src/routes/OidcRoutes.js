import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { NotAuthenticated, NotAuthorized } from '../default-component';
import { getPath } from './route-utils';
import { SilentCallback } from '../callbacks';

const propTypes = {
  notAuthenticated: PropTypes.node,
  notAuthorized: PropTypes.node,
  callbackComponent: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
  callbackComponentOverride: PropTypes.node,
  configuration: PropTypes.shape({
    redirect_uri: PropTypes.string.isRequired,
    silent_redirect_uri: PropTypes.string.isRequired,
  }).isRequired,
  children: PropTypes.node,
};

const defaultProps = {
  notAuthenticated: null,
  notAuthorized: null,
  children: null,
  callbackComponentOverride: null,
};

const OidcRoutes = ({
  notAuthenticated,
  notAuthorized,
  callbackComponent: CallbackComponent,
  callbackComponentOverride,
  configuration,
  children,
}) => {
  const [path, setPath] = useState(window.location.pathname);

  const setNewPath = () => setPath(window.location.pathname);
  useEffect(() => {
    setNewPath();
    window.addEventListener('popstate', setNewPath, false);
    return () => window.removeEventListener('popstate', setNewPath, false);
  });

  const NotAuthenticatedComponent = notAuthenticated || NotAuthenticated;
  const NotAuthorizedComponent = notAuthorized || NotAuthorized;
  const silentCallbackPath = getPath(configuration.silent_redirect_uri);
  const callbackPath = getPath(configuration.redirect_uri);

  switch (path) {
    case callbackPath:
      return <CallbackComponent callbackComponentOverride={callbackComponentOverride} />;
    case silentCallbackPath:
      return <SilentCallback />;
    case '/authentication/not-authenticated':
      return <NotAuthenticatedComponent />;
    case '/authentication/not-authorized':
      return <NotAuthorizedComponent />;
    default:
      return <>{children}</>;
  }
};

OidcRoutes.propTypes = propTypes;
OidcRoutes.defaultProps = defaultProps;

export default React.memo(OidcRoutes);
