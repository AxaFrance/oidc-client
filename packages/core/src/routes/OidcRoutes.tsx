import React, { ComponentType, FC, PropsWithChildren, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { UserManagerSettings } from 'oidc-client';
import { NotAuthenticated, NotAuthorized, SessionLost } from '../default-component';
import { getPath } from './route-utils';
import { SilentCallback } from '../callbacks';

const propTypes = {
  notAuthenticated: PropTypes.elementType,
  notAuthorized: PropTypes.elementType,
  callbackComponent: PropTypes.elementType.isRequired,
  configuration: PropTypes.shape({
    redirect_uri: PropTypes.string,
    popup_redirect_uri: PropTypes.string,
    popupWindowFeatures: PropTypes.string,
    silent_redirect_uri: PropTypes.string.isRequired,
  }).isRequired,
  children: PropTypes.node,
};

const defaultProps: Partial<OidcRoutesProps> = {
  notAuthenticated: null,
  notAuthorized: null,
};

type OidcRoutesProps = {
  notAuthenticated?: ComponentType;
  notAuthorized?: ComponentType;
  callbackComponent: ComponentType;
  sessionLost?: ComponentType;
  configuration: UserManagerSettings;
};

const OidcRoutes: FC<PropsWithChildren<OidcRoutesProps>> = ({
  notAuthenticated,
  notAuthorized,
  callbackComponent: CallbackComponent,
  sessionLost,
  configuration,
  children,
}) => {
  const [path, setPath] = useState(window.location.pathname);

  const setNewPath = () => setPath(window.location.pathname);
  useEffect(() => {
    setNewPath();
    window.addEventListener('popstate', setNewPath, false);
    return () => window.removeEventListener('popstate', setNewPath, false);
  });

  const NotAuthenticatedComponent = notAuthenticated || NotAuthenticated;
  const NotAuthorizedComponent = notAuthorized || NotAuthorized;
  const SessionLostComponent = sessionLost || SessionLost;
  const silentCallbackPath = getPath(configuration.silent_redirect_uri);
  var callbackPath = null;
  if(configuration.redirect_uri !== undefined && configuration.redirect_uri !== null){
    callbackPath = getPath(configuration.redirect_uri);
  }else if(configuration.popup_redirect_uri !== undefined && configuration.popup_redirect_uri !== null){
    callbackPath = getPath(configuration.popup_redirect_uri);
  }else{
    throw TypeError("redirect_uri or popup_redirect_uri have to be set in the configuration!")
  }
   

  switch (path) {
    case callbackPath:
      return <CallbackComponent />;
    case silentCallbackPath:
      return <SilentCallback />;
    case '/authentication/not-authenticated':
      return <NotAuthenticatedComponent />;
    case '/authentication/not-authorized':
      return <NotAuthorizedComponent />;
    case '/authentication/session-lost':
      return <SessionLostComponent />;
    default:
      return <>{children}</>;
  }
};

// @ts-ignore
OidcRoutes.propTypes = propTypes;
OidcRoutes.defaultProps = defaultProps;

export default React.memo(OidcRoutes);
