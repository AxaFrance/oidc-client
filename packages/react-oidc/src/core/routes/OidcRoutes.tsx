import {getPath, ILOidcLocation} from '@axa-fr/oidc-client';
import React, { ComponentType, FC, PropsWithChildren, useEffect, useState } from 'react';

import CallbackComponent from '../default-component/Callback.component.js';
import SilentCallbackComponent from '../default-component/SilentCallback.component.js';
import SilentLoginComponent from '../default-component/SilentLogin.component.js';
import { CustomHistory } from './withRouter.js';

type OidcRoutesProps = {
  configurationName: string;
  redirect_uri: string;
  silent_redirect_uri?: string;
  silent_login_uri?: string;
  withCustomHistory?: () => CustomHistory;
};

const OidcRoutes: FC<PropsWithChildren<OidcRoutesProps>> = ({
  redirect_uri,
  silent_redirect_uri,
  silent_login_uri,
  children, 
                                                              configurationName,
  withCustomHistory = null,
}) => {
  // This exist because in next.js window outside useEffect is null
  const pathname = window ? getPath(window.location.href) : '';

  const [path, setPath] = useState(pathname);

  useEffect(() => {
    const setNewPath = () => setPath(getPath(window.location.href));
    setNewPath();
    window.addEventListener('popstate', setNewPath, false);
    return () => window.removeEventListener('popstate', setNewPath, false);
  }, []);

  const callbackPath = getPath(redirect_uri);

  if (silent_redirect_uri) {
    if (path === getPath(silent_redirect_uri)) {
      return <SilentCallbackComponent configurationName={configurationName} />;
    }
  }

  if (silent_login_uri) {
    if (path === getPath(silent_login_uri)) {
      return <SilentLoginComponent configurationName={configurationName} />;
    }
  }

  switch (path) {
    case callbackPath:
      return <CallbackComponent configurationName={configurationName} withCustomHistory={withCustomHistory} >{children}</CallbackComponent>;
    default:
      return <>{children}</>;
  }
};

export default React.memo(OidcRoutes);
