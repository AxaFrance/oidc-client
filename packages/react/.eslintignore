build
bin
node_modules
dist
**/*.spec.ts
**/*.spec.tsx
**/*.test.ts
**/*.test.tsx
**/*.stories.tsx
./src/setupTests.ts
setupTests.js
.eslintrc.js
config
jest.config.js
App.specold.tsx
