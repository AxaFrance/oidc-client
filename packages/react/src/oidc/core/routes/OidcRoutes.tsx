import React, { ComponentType, FC, PropsWithChildren, useEffect, useState } from 'react';
import { getPath } from './route-utils';
import CallbackComponent from '../default-component/Callback.component';
import SilentCallbackComponent from "../default-component/SilentCallback.component";
import ServiceWorkerInstall from "../default-component/ServiceWorkerInstall.component";
import { CustomHistory } from "./withRouter";

const defaultProps: Partial<OidcRoutesProps> = {

};

type OidcRoutesProps = {
  callbackSuccessComponent?: ComponentType;
  callbackErrorComponent?: ComponentType;
  authenticatingComponent?: ComponentType;
  configurationName:string;
  redirect_uri: string;
  silent_redirect_uri?: string;
  withCustomHistory?: () => CustomHistory;
};

const OidcRoutes: FC<PropsWithChildren<OidcRoutesProps>> = ({
  callbackErrorComponent,
  callbackSuccessComponent,
                                                              authenticatingComponent,  
                                                              redirect_uri,
                                                              silent_redirect_uri,
  children, configurationName,
  withCustomHistory=null,
}) => {
  // This exist because in next.js window outside useEffect is null
  let pathname = window ? getPath(window.location.href) : '';
  
  const [path, setPath] = useState(pathname);
  
  useEffect(() => {
    const setNewPath = () => setPath(getPath(window.location.href));
    setNewPath();
    window.addEventListener('popstate', setNewPath, false);
    return () => window.removeEventListener('popstate', setNewPath, false);
  }, []);
  
  const callbackPath = getPath(redirect_uri);

  if(silent_redirect_uri){
    if(path === getPath(silent_redirect_uri)){
      return <SilentCallbackComponent configurationName={configurationName} />
    }
  }

  switch (path) {
    case callbackPath:
      return <CallbackComponent callBackError={callbackErrorComponent} callBackSuccess={callbackSuccessComponent} configurationName={configurationName} withCustomHistory={withCustomHistory} />;
    case callbackPath +"/service-worker-install" :
      return <ServiceWorkerInstall callBackError={callbackErrorComponent} authenticating={authenticatingComponent} configurationName={configurationName} />;  
    default:
      return <>{children}</>;
  }
};

// @ts-ignore
OidcRoutes.defaultProps = defaultProps;

export default React.memo(OidcRoutes);
