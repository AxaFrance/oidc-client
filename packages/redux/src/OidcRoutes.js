import React from 'react';
import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import AuthenticationRoutes from './AuthenticationRoutes';

const propTypes = {
  notAuthentified: PropTypes.node,
  notAuthorized: PropTypes.node,
  children: PropTypes.node,
};

const defaultProps = {
  notAuthentified: null,
  notAuthorized: null,
  children: PropTypes.node,
};

const OidcRoutes = ({ notAuthentified, notAuthorized, children }) => (
  <Switch>
    <Route
      path="/authentication"
      component={AuthenticationRoutes(notAuthentified, notAuthorized)}
    />
    <Route render={() => children} />
  </Switch>
);

OidcRoutes.propTypes = propTypes;
OidcRoutes.defaultProps = defaultProps;

export default OidcRoutes;
