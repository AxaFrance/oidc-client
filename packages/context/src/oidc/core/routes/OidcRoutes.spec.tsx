import * as React from 'react';
import renderer from 'react-test-renderer';
import OidcRoutes from './OidcRoutes';
import { FC } from 'react';

describe('Authenticating test suite', () => {
  it('renders correctly', () => {
    const notAuthenticated: FC = () => <>'notAuthenticated'</>;
    const notAuthorized: FC = () => <>'notAuthorized'</>;
    const props = {
      children: 'http://url.com',
      notAuthenticated,
      notAuthorized,
      callbackComponent: () => <div>tcallback component</div>,
      configuration: {
        redirect_uri: 'http://example.com:3000/authentication/callback',
        silent_redirect_uri: 'http://example.com:3000/authentication/silent_callback',
      },
    };
    const tree = renderer.create(<OidcRoutes {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
