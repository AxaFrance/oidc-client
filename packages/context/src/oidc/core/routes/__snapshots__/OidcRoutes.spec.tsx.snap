// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticating test suite renders correctly 1`] = `"http://url.com"`;
